### PR TITLE
CAS-1033 MessageReplyView style

### PR DESCRIPTION
**Pull request by samiuelson**
https://stream-io.atlassian.net/browse/CAS-1033

### 🎯 Goal

Allow customizing MessageReplyView

### 🛠 Implementation details

* Created `MessageReplyStyle` integrated with `TransformStyle` API
* Added `MessageListView` XML attrs
* Fixed mine/theirs logic in `ReplyDecorator`

### 🧪 Testing

Customize MessageReply using TransformStyle or XML attrs and check out the look of message list item which is a reply.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt [hunk 1]
```diff
@@ -99,35 +100,48 @@ internal class MessageReplyView : FrameLayout {
...
+                    val tintColor = style?.messageBackgroundColorTheirs ?: context.getColorCompat(R.color.stream_ui_white)
+                    setTint(tintColor)
                 }
             }
         }
     }
 
+    private fun isLink(message: Message) = message.attachments
+        .lastOrNull()
+        ?.type == ModelType.attach_link
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:143]: Why a message is a link only when the last attachment is of type link? If the first is a link, then the message is not a link?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt [hunk 2]
```diff
@@ -99,35 +100,48 @@ internal class MessageReplyView : FrameLayout {
...
                 }
                 isMine -> {
-                    paintStyle = Paint.Style.FILL
-                    setTint(context.getColorCompat(R.color.stream_ui_grey_whisper))
+                    paintStyle = Paint.Style.FILL_AND_STROKE
+                    val color =
+                        style?.messageBackgroundColorMine ?: context.getColorCompat(R.color.stream_ui_grey_whisper)
+                    setTint(color)
+                    style?.messageStrokeColorMine?.let {
+                        setStrokeTint(it)
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt:126]: You can use function reference here 💇  💅

**Review by leandroBorgesFerreira — COMMENTED** [review]
2 comments

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt (hunk 1 above)
Comment (samiuelson) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt]: Fixed

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageReplyStyle.kt [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+ */
+public data class MessageReplyStyle(
+    @ColorInt
+    public val messageBackgroundColorMine: Int?,
+    @ColorInt
+    public val messageBackgroundColorTheirs: Int?,
+    @ColorInt
+    public val linkBackgroundColorMine: Int?,
+    @ColorInt
+    public val linkBackgroundColorTheirs: Int?,
```
Comment (adasiewiczr) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageReplyStyle.kt:48]: Why these parameters are nullable?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/internal/MessageInputFieldView.kt [hunk 4]
```diff
@@ -235,7 +235,8 @@ internal class MessageInputFieldView : FrameLayout {
         switchToMessageMode()
         binding.messageReplyView.setMessage(
             currentMode.repliedMessage,
-            ChatDomain.instance().currentUser.id == currentMode.repliedMessage.user.id
+            ChatDomain.instance().currentUser.id == currentMode.repliedMessage.user.id,
+            null,
```
Comment (adasiewiczr) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/internal/MessageInputFieldView.kt:239]: `MessageInputView` should have a separate style, right?

Comment (samiuelson) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/internal/MessageInputFieldView.kt:239]: Yes

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageReplyStyle.kt (hunk 3 above)
Comment (samiuelson) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageReplyStyle.kt:48]: I'll make those props non null to keep consistency with other styles

Comment (samiuelson) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageReplyStyle.kt]: Fixed

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**